### PR TITLE
Push docker CI images to repo specific docker registry

### PR DESCRIPTION
**Pull request by feelepxyz**
Fix actions ci on forks.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,34 +16,26 @@ jobs:
...
             --cache-from "dependabot/dependabot-core:latest" \
             --cache-from "dependabot/dependabot-core-ci:latest" \
             .
       - name: Push dependabot-core-ci image to CI cache repo
         run: |
-          docker push "dependabot/dependabot-core-ci:latest"
-          docker push "dependabot/dependabot-core-ci:${{ github.sha }}"
+          docker login docker.pkg.github.com -u x -p ${{ secrets.GITHUB_TOKEN }}
+          docker push "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}"
+          docker push "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:latest"
```
Comment (feelepxyz) [inline .github/workflows/ci.yml:33]: Do we also need this?
```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,34 +16,25 @@ jobs:
...
-          docker push "dependabot/dependabot-core-ci:latest"
-          docker push "dependabot/dependabot-core-ci:${{ github.sha }}"
+          docker login docker.pkg.github.com -u x -p ${{ secrets.GITHUB_TOKEN }}
+          docker push "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}"
 
   rubocop:
     name: Rubocop
     runs-on: ubuntu-latest
-    container: dependabot/dependabot-core-ci:${{ github.sha }}
+    container: docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}
```
Comment (feelepxyz) [inline .github/workflows/ci.yml:37]: looks like we need to login here to make this work 😕

Comment (feelepxyz) [reply .github/workflows/ci.yml:37]: tested on a branch here https://github.com/feelepxyz/dependabot-core/runs/919035546?check_suite_focus=true

Comment (feelepxyz) [reply .github/workflows/ci.yml:37]: https://github.community/t/is-it-possible-to-login-to-dockerhub-to-initialize-containers-in-github-actions/16263/3 😓

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -19,44 +16,43 @@ jobs:
...
+        uses: actions/download-artifact@v2
+        with:
+          name: dependabot-core-ci-image.tar
+      - name: Restore image
+        run: |
+          docker image import dependabot-core-ci-image.tar "dependabot/dependabot-core-ci:${{ github.sha }}"
       - name: Run Rubocop linting
         run: |
-          cd /home/dependabot/dependabot-core && rake ci:rubocop
+          docker run --rm dependabot/dependabot-core:${{ github.sha }} bash -c "cd /home/dependabot/dependabot-core && rake ci:rubocop"
```
Comment (feelepxyz) [inline .github/workflows/ci.yml:50]:
```suggestion
          docker run --rm dependabot/dependabot-core-ci:${{ github.sha }} bash -c "cd /home/dependabot/dependabot-core && rake ci:rubocop"
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -19,44 +16,43 @@ jobs:
...
-    container: dependabot/dependabot-core-ci:${{ github.sha }}
     needs: ci-image
     steps:
+      - name: Load docker image from artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: dependabot-core-ci-image.tar
+      - name: Restore image
+        run: |
+          docker image import dependabot-core-ci-image.tar "dependabot/dependabot-core-ci:${{ github.sha }}"
```
Comment (localheinz) [inline .github/workflows/ci.yml:47]: Perhaps this helps?


```suggestion
          docker load < dependabot-core-ci-image.tar
```

For reference, see https://pspdfkit.com/blog/2019/docker-import-export-vs-load-save/.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -19,44 +16,43 @@ jobs:
...
-      - name: Push dependabot-core-ci image to CI cache repo
-        run: |
-          docker push "dependabot/dependabot-core-ci:latest"
-          docker push "dependabot/dependabot-core-ci:${{ github.sha }}"
+          docker image save -o dependabot-core-ci-image.tar "dependabot/dependabot-core-ci:${{ github.sha }}"
+      - name: Save image as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: dependabot-core-ci-image.tar
+          path: dependabot-core-ci-image.tar
```
Comment (feelepxyz) [inline .github/workflows/ci.yml:34]: Looks like this step is taking like 8-9 minutes 😯 testing the github packages approach again: https://github.com/feelepxyz/dependabot-core/actions

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -21,219 +18,93 @@ jobs:
...
-            -t "dependabot/dependabot-core:latest" \
-            -t "dependabot/dependabot-core:${{ env.VERSION }}" \
-            --cache-from ubuntu:18.04 \
-            --cache-from "dependabot/dependabot-core:latest" \
-            .
       - name: Build dependabot-core-ci image
         run: |
           rm .dockerignore
           docker build \
             -t "dependabot/dependabot-core-ci:${{ github.sha }}" \
```
Comment (feelepxyz) [inline .github/workflows/ci.yml:25]:
```suggestion
            -t "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}" \
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -21,219 +18,93 @@ jobs:
...
-      - name: Push dependabot-core-ci image to CI cache repo
-        run: |
-          docker push "dependabot/dependabot-core-ci:latest"
-          docker push "dependabot/dependabot-core-ci:${{ github.sha }}"
+          docker image save -o dependabot-core-ci-image.tar "dependabot/dependabot-core-ci:${{ github.sha }}"
+      - name: Save image as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: dependabot-core-ci-image.tar
+          path: dependabot-core-ci-image.tar
```
Comment (feelepxyz) [inline .github/workflows/ci.yml:36]:
```suggestion
      - name: Push ci image
         run: |
           docker login docker.pkg.github.com -u x -p ${{ secrets.GITHUB_TOKEN }}
           docker push "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}"
```

File: .github/workflows/ci.yml (hunk 5 above)
Comment (localheinz) [reply .github/workflows/ci.yml:36]: 👍

**Pull request closed without merging**